andela/ah-frontend-zeus
# #162163141 Enable registered users to follow/unfollow each other

### What does this PR do?
Enables logged in users to follow and unfollow other users, and to see a list of followers and profiles followed by them
#### Description of the task to be completed?
- Add action types for following and unfollowing users
- Write reducers for the actions to follow and unfollow
- Update the state shape to hold required data
- Implement thunks to dispatch actions to the reducers
- Build static UI representation to follow and unfollow users
- Connect UI to the Redux state for dynamic behaviour
- Write tests
- Refactor codebase to meet style guidelines
#### How should this be manually tested?
- On this branch, Login to Authors Haven
- Click the **_Users_** link in the top right drop-down menu to see a list of all users
- Click the button alongside each profile to follow/unfollow that profile
- To see a list of followers or profiles you are following:
  - Click the **_Profile_** link from the top right drop-down menu to see your profile
  - Click the following/followers link to view the respective profiles
#### Screenshot
![image](https://user-images.githubusercontent.com/34346070/51699730-8ee52380-201e-11e9-88b8-d8dbf56e6f23.png)
#### What are the relevant Pivotal Tracker stories?
[#162163141](https://www.pivotaltracker.com/story/show/162163141)
